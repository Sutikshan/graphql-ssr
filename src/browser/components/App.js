import React from "react";
import { graphql } from "react-apollo";
import gql from "graphql-tag";

function App(graphqlResponse) {
  if (graphqlResponse.data.loading) {
    return <div>Loading</div>;
  }

  const { carOfTheWeek } = graphqlResponse.data;

  return (
    <div>
      <h1>Car of the Week</h1>
      <div className="home-page">
        <img className="car-image" src={carOfTheWeek.imageUrl} />
        <div className="review-text">{carOfTheWeek.review}</div>
        <button className="search-button">Search Catalog</button>
      </div>
    </div>
  );
}

export default graphql(
  gql`
    query AppQuery {
      carOfTheWeek {
        id
        name
        imageUrl
        review
      }
    }
  `
)(App);
