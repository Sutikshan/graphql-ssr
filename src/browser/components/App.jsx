import React from "react";
import { graphql } from "react-apollo";
import { Link } from "react-router-dom";
import { carOfTheWeekQuery } from "./queries";

export function App(graphqlResponse) {
  let { carOfTheWeek, loading } = graphqlResponse.data;

  if (loading) {
    carOfTheWeek = {
      imageUrl: undefined,
      review: "loading"
    };
  }

  return (
    <div>
      <h1>Car of the Week</h1>
      <div className="detail-page">
        <img
          className="car-image"
          alt="loading..."
          src={carOfTheWeek.imageUrl}
        />
        <div className="review-text">{carOfTheWeek.review}</div>
        <Link to="/search">
          <button className="search-button">Search Catalog</button>
        </Link>
      </div>
    </div>
  );
}

export default graphql(carOfTheWeekQuery)(App);
