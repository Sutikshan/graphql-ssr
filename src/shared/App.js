import React, { Component } from "react";
import PropTypes from "prop-types";

class App extends Component {
  render() {
    return (
      <div>
        <h1>Car of the Week</h1>
        <div className="home-page">
          <img className="car-image" src={this.props.carOfTheWeek.imageUrl} />
          <div className="review-text">{this.props.carOfTheWeek.review}</div>
          <button className="search-button">Search Catalog</button>
        </div>
      </div>
    );
  }
}

App.propTypes = {
  carOfTheWeek: PropTypes.object
};
export default App;
