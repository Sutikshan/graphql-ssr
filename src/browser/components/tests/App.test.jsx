import React from "react";
import { shallow } from "enzyme";
import { App } from "../App";

it("should render a App component correctly", () => {
  const data = {
    carOfTheWeek: { imageUrl: "http://rediff.com/image.png", Name: "Platanito" }
  };
  const wrapper = shallow(<App data={data} />);

  expect(wrapper).toMatchSnapshot();
});
