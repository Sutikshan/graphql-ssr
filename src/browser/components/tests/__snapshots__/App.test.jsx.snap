// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a App component correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    data={
      Object {
        "carOfTheWeek": Object {
          "Name": "Platanito",
          "imageUrl": "http://rediff.com/image.png",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Car of the Week
        </h1>,
        <div
          className="detail-page"
        >
          <img
            alt="loading..."
            className="car-image"
            src="http://rediff.com/image.png"
          />
          <div
            className="review-text"
          />
          <Link
            replace={false}
            to="/search"
          >
            <button
              className="search-button"
            >
              Search Catalog
            </button>
          </Link>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Car of the Week",
        },
        "ref": null,
        "rendered": "Car of the Week",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="loading..."
              className="car-image"
              src="http://rediff.com/image.png"
            />,
            <div
              className="review-text"
            />,
            <Link
              replace={false}
              to="/search"
            >
              <button
                className="search-button"
              >
                Search Catalog
              </button>
            </Link>,
          ],
          "className": "detail-page",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "loading...",
              "className": "car-image",
              "src": "http://rediff.com/image.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "review-text",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <button
                className="search-button"
              >
                Search Catalog
              </button>,
              "replace": false,
              "to": "/search",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Search Catalog",
                "className": "search-button",
              },
              "ref": null,
              "rendered": "Search Catalog",
              "type": "button",
            },
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Car of the Week
          </h1>,
          <div
            className="detail-page"
          >
            <img
              alt="loading..."
              className="car-image"
              src="http://rediff.com/image.png"
            />
            <div
              className="review-text"
            />
            <Link
              replace={false}
              to="/search"
            >
              <button
                className="search-button"
              >
                Search Catalog
              </button>
            </Link>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Car of the Week",
          },
          "ref": null,
          "rendered": "Car of the Week",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="loading..."
                className="car-image"
                src="http://rediff.com/image.png"
              />,
              <div
                className="review-text"
              />,
              <Link
                replace={false}
                to="/search"
              >
                <button
                  className="search-button"
                >
                  Search Catalog
                </button>
              </Link>,
            ],
            "className": "detail-page",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "loading...",
                "className": "car-image",
                "src": "http://rediff.com/image.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "review-text",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <button
                  className="search-button"
                >
                  Search Catalog
                </button>,
                "replace": false,
                "to": "/search",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Search Catalog",
                  "className": "search-button",
                },
                "ref": null,
                "rendered": "Search Catalog",
                "type": "button",
              },
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
